rotkehlchen/chain/ethereum/modules/uniswap/utils.py

```diff
@@ -253,3 +268,220 @@ def historical_uniswap_v2_lp_price(
         token_price_func_args=[to_asset, timestamp],
         block_identifier=block_identifier,
     )
+
+
+def uniswap_v3_lp_token_balances(
+    userdb: 'DBHandler',
+    address: ChecksumEthAddress,
+    ethereum: 'EthereumManager',
+    known_assets: Set[EthereumToken],
+    unknown_assets: Set[EthereumToken],
+) -> List[NFTLiquidityPool]:
+    nft_manager_contract = EthereumContract(
+        address=UNISWAP_NFT_MANAGER_ADDRESS,
+        abi=UNISWAP_V3_NFT_MANAGER_ABI,
+        deployed_block=UNISWAP_V3_DEPLOYED_BLOCK,
+    )
+    address = to_checksum_address(address)
+    my_positions = nft_manager_contract.call(
+        ethereum=ethereum,
+        method_name="balanceOf",
+        arguments=[address],
+    )
+    balances = []
+    if my_positions > 0:
+        chunks = list(get_chunks(list(range(my_positions)), n=10))
+        for chunk in chunks:
+            tokens_ids_multicall = multicall_2(
+                ethereum=ethereum,
+                require_success=True,
+                calls=[
+                    (
+                        UNISWAP_NFT_MANAGER_ADDRESS,
+                        nft_manager_contract.encode('tokenOfOwnerByIndex', [address, index]),
+                    )
+                    for index in chunk
+                ],
+            )
+            tokens_ids = [
+                nft_manager_contract.decode(   # pylint: disable=unsubscriptable-object
+                    result=data[1],
+                    method_name='tokenOfOwnerByIndex',
+                    arguments=[address, index],
+                )[0]
+                for index, data in enumerate(tokens_ids_multicall)
+            ]
+            positions_multicall = multicall_2(
+                ethereum=ethereum,
+                require_success=True,
+                calls=[
+                    (
+                        UNISWAP_NFT_MANAGER_ADDRESS,
+                        nft_manager_contract.encode('positions', [token_id]),
+                    )
+                    for token_id in tokens_ids
+                ],
+            )
+            positions = [
+                nft_manager_contract.decode(
+                    result=data[1],
+                    method_name='positions',
+                    arguments=[tokens_ids[index]],
+                )
+                for index, data in enumerate(positions_multicall)
+            ]
+            pool_addresses = [
+                compute_pool_address(
+                    factory_address=UNISWAP_V3_FACTORY_ADDRESS,
+                    token0_address=position[2],
+                    token1_address=position[3],
+                    fee=position[4],
+                )
+                for position in positions
+            ]
```

Comment:
Is the address always guaranteed to exists? I believe so since you are iterating over positions for pools that have been returned by the manager